src/cli.rs

```diff
@@ -544,14 +544,16 @@ async fn run_deploy(
         let mut deploy_defs = deploy_data.defs()?;
 
         if node.generic_settings.interactive_sudo.unwrap_or(false) {
-            info!("Interactive sudo is enabled, you will be prompted for the sudo password for {}.", node.node_settings.hostname);
+            warn!("Interactive sudo is enabled! Using a sudo password is less secure than correctly configured SSH keys.\nPlease use keys in production environments.");
+            info!("You will now be prompted for the sudo password for {}.", node.node_settings.hostname);
             let sudo_password = rpassword::prompt_password(format!("(sudo for {}) Password: ", node.node_settings.hostname)).unwrap_or("".to_string());
 
             deploy_defs.sudo_password = Some(sudo_password);
-        
+            
+            // this instructs sudo to hide the password prompt and accept input from stdin
             if let Some(original) = deploy_defs.sudo.clone()  {
                 if !original.contains("-S") {
-                    deploy_defs.sudo = Some(format!("{} -S -p \"\"", original));
+                    deploy_defs.sudo = Some(format!("{} -S -p \"\"", original)); 
```

Comment:
This piece of diff looks suspicious :sweat_smile:

Reply:
An additional concern I have regarding this line is that alternative some alternative `sudo` commands may not support provided flags, e.g. `doas`. I'm not sure what is the best way to fix this, perhaps, if we use custom `sudo` attribute together with `interactiveSudo = true;` we should require updating `sudo` attribute provide required flags instead of hardcoding them here

Reply:
Yeah I thought this may be too bold to assume...

Do you think maybe the thing to do is:
If no custom `sudo` attribute is defined, leave this code as-is. If the user has defined a custom `sudo` attribute, print a message along the lines of "ensure your sudo command can accept the password from stdin"? Or if theres already too many messages, wait until deployment fails and print a message like "you're using a custom `sudo` and `interactiveSudo`. Ensure your custom sudo can accept the password from stdin for this to work."

Reply:
Yes, such an approach sounds good to me

Reply:
I think it's better to print the warning before, because the deployment may fail due to a different error and the message after might potentially be misleading